### PR TITLE
implement ignore_missing_dates argument and respective warnings when …

### PR DESCRIPTION
**Pull request by e-kotov**
…user selects out of bounds dates

To test

```r
devtools::load_all()

# should stop with error because of overlap between v1 and v2
overlapping_dates <- c("2021-03-01", "2023-03-01")
x <- spod_get(type = "od", zones = "distr", dates = overlapping_dates)

# set dates that are all outside valid v1 dates
fully_missing_v1_dates <- c("2021-09-01", "2021-09-02")

# should stop with error because of missing v1 dates
x <- spod_get(type = "od", zones = "distr", dates = fully_missing_v1_dates)

# should stop with error, as we set only dates that are completely outside the requested range
x <- spod_get(type = "od", zones = "distr", dates = fully_missing_v1_dates, ignore_missing_dates = TRUE)

# set v1 dates where some are missing and some are valid
partially_missing_v1_dates <- c("2021-03-01", "2021-09-01")

# should stop with error because of missing v1 dates
x <- spod_get(type = "od", zones = "distr", dates = partially_missing_v1_dates)

# should proceed ignoring missing v1 dates
x <- spod_get(type = "od", zones = "distr", dates = partially_missing_v1_dates, ignore_missing_dates = TRUE)
spod_disconnect(x) # disconnect

# set v2 dates where some are missing completely
fully_missing_v2_dates <- c("2026-09-01", "2026-09-02")

# should stop with error because of missing v2 dates
x <- spod_get(type = "od", zones = "distr", dates = fully_missing_v2_dates)

# set partially missing v2 dates
partially_missing_v2_dates <- c("2024-03-01", "2026-09-01")

# should stop with error because of missing v2 dates
x <- spod_get(type = "od", zones = "distr", dates = partially_missing_v2_dates)

# should proceed ignoring missing v2 dates
x <- spod_get(type = "od", zones = "distr", dates = partially_missing_v2_dates, ignore_missing_dates = TRUE)
spod_disconnect(x)

```

### CODE REVIEW

**Comment by Robinlovelace** [comment]
Hi @e-kotov thanks for this, I'm catching up on various things on my first full week back after paternity leave. Will take a look at this right now...

File: R/convert_data.R [hunk 1]
```diff
@@ -41,15 +41,16 @@ spod_convert <- function(
   quiet = FALSE,
   max_mem_gb = max(4, spod_available_ram() - 4),
   max_n_cpu = parallelly::availableCores() - 1,
-  max_download_size_gb = 1
+  max_download_size_gb = 1,
+  ignore_missing_dates = FALSE
```
Comment (Robinlovelace) [inline R/convert_data.R:45]: 👍 error is a good default.

**Review by Robinlovelace — APPROVED** [review]
Looks great to me. I have not run the code but it looks great to me, and the mini examples in the PR description give me confidence this runs. Great work Egor!

File: R/convert_data.R (hunk 1 above)
Comment (e-kotov) [reply R/convert_data.R:45]: thanks for catching this!

Comment (Robinlovelace) [reply R/convert_data.R:45]: What did I catch?

Comment (e-kotov) [reply R/convert_data.R:45]: ahh, i though I wrongly defaulted it to TRUE. I misinterpreted this comment as a comment to your commit correcting me) Anyway, yes, good to error by default, as this is extremely important for aggregations that are dependent on number of non-missing days in a month

**Comment by e-kotov** [comment]
@Robinlovelace thanks for taking the time to review on your first week back to work! merging now and moving on to the last bit before the CRAN submission - safety checks for all user facing functions

**Comment by Robinlovelace** [comment]
Not quite first day back, first day of the first full week back, to be specific! Great work, thank you for forging ahead on this, I will pick up on the paper ASAP.

**Pull request merged**